src/Domain/Services/ExtendRestApi.php

```diff
@@ -0,0 +1,135 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Domain\Services;
+
+use Throwable;
+
+/**
+ * Service class to provide utility functions to extend REST API.
+ */
+class ExtendRestApi {
+	/**
+	 * Valid endpoints to extend
+	 *
+	 * @var array
+	 */
+	private $endpoints = [ 'item' ];
```

Comment:
Each time we add a new integration point, we would add it here.
Integration points should have a slug, this would keep them controlled and would allow us to easily depreciate them.

Reply:
What does `'item'` refer to here and how did you derive the name `item`?

Reply:
It works like the suggested `$valid_identifiers`, I derived it from the endpoint name, but I think I can use the route name you pointed to.